csup/dict.go

```diff
@@ -87,7 +84,7 @@ func (d *DictEncoder) encodeValues(group *errgroup.Group) {
 	for key, tag := range d.tags {
 		byteSlices[tag] = zcode.Bytes(key)
 	}
-	d.values.(resetter).reset()
+	d.values = newPrimitiveEncoder(d.typ)
```

Comment:
Do you need this given that d.values is already initialized with the encoder passed to NewDictEncoder?

Reply:
Yes because current values is not deduplicated so therefore needs to be rewritten.

Reply:
Ah, I get it now.